sdk/src/main/scala/com/horizen/network/SidechainNodeViewSynchronizer.scala

```diff
@@ -45,8 +56,173 @@ class SidechainNodeViewSynchronizer(networkControllerRef: ActorRef,
       }
   }
 
+  /**
+   * Method copied from the parent class with the addition of isReindexing check
+   *
+   * Object ids coming from other node.
+   * Filter out modifier ids that are already in process (requested, received or applied),
+   * request unknown ids from peer and set this ids to requested state.
+   */
+  override protected def processInv(invData: InvData, peer: ConnectedPeer): Unit = {
+    if (this.isReindexing){
+      log.warn("Got data from peer while reindexing - will be discarded")
+    }else {
+      (mempoolReaderOpt, historyReaderOpt) match {
+        case (Some(mempool), Some(history)) =>
+          val modifierTypeId = invData.typeId
+          val newModifierIds = (modifierTypeId match {
+            case Transaction.ModifierTypeId =>
+              invData.ids.filter(mid => deliveryTracker.status(mid, mempool) == ModifiersStatus.Unknown)
+            case _ =>
+              invData.ids.filter(mid => deliveryTracker.status(mid, history) == ModifiersStatus.Unknown)
+          })
+            .take(deliveryTracker.getPeerLimit(peer))
+
+          if (newModifierIds.nonEmpty) {
+            val msg = Message(requestModifierSpec, Right(InvData(modifierTypeId, newModifierIds)), None)
+            peer.handlerRef ! msg
+            deliveryTracker.setRequested(newModifierIds, modifierTypeId, peer)
+          }
+
+        case _ =>
+          log.warn(s"Got data from peer while readers are not ready ${(mempoolReaderOpt, historyReaderOpt)}")
+      }
+    }
+  }
+
+
+
+  /**
+   * Method copied from the parent class with the addition of isReindexing check
+   *
+   * Logic to process modifiers got from another peer.
+   * Filter out non-requested modifiers (with a penalty to spamming peer),
+   * parse modifiers and send valid modifiers to NodeViewHolder
+   */
+  override protected def modifiersFromRemote(data: ModifiersData, remote: ConnectedPeer): Unit = {
```

Comment:
removed the check here, but had to keep the method because I changed the sub-called validateAndSetStatus
(we need to report the reception of the modifier anyway, to avoid peer penalization)
Note: we can add a ticket to update method modifiers in Sparkz from private to protected, and later clean a bit the code